src/proxy/MapProxy.ts

```diff
@@ -838,8 +829,16 @@ export class MapProxy<K, V> extends BaseProxy implements IMap<K, V> {
     }
 
     protected removeAllInternal(predicate: Predicate): Promise<void> {
-        const request = MapRemoveAllCodec.encodeRequest(this.name, this.toData(predicate));
-        return this.encodeInvokeOnRandomTarget(MapRemoveAllCodec, () => {}, request);
+        let predicateData: Data;
+        try {
+            predicateData = this.toData(predicate);
+        } catch (e) {
+            if (e instanceof SchemaNotReplicatedError) {
+                return this.registerSchema(e.schema, e.clazz).then(() => this.removeAllInternal(predicate));
```

Comment:
There is a warning about coverage on this line. This is because you did not test removeAll method with compact. You can add removeAll method to `CompactPublicAPIsTest`. Note that, you need to use `CompactPredicate()` as the predicate.

Reply:
You can add new test here, under "remove"

 
![image](https://user-images.githubusercontent.com/32355782/188657095-f96e9cad-643c-4e1b-b780-8e914566b1b9.png)